packages/fiberplane-hono/src/routes/api/auth.ts

```diff
@@ -0,0 +1,100 @@
+import { type Env, Hono } from "hono";
+import { deleteCookie, getCookie, setCookie } from "hono/cookie";
+import { HTTPException } from "hono/http-exception";
+import type { FetchFn, FiberplaneAppType } from "../../types";
+
+export default function createAuthApiRoute<E extends Env>(
+  apiKey: string,
+  fetchFn: FetchFn,
+  fiberplaneServicesUrl: string,
+) {
+  const app = new Hono<E & FiberplaneAppType<E>>();
+
+  app.get("/authorize", async (c) => {
+    const { embeddedUrl } = await getApp(
+      apiKey,
+      fetchFn,
+      fiberplaneServicesUrl,
+    );
+
+    return c.redirect(
+      `${fiberplaneServicesUrl}/api/auth/authorize?embeddedUrl=${encodeURIComponent(embeddedUrl)}`,
+    );
+  });
+
+  app.get("/callback", async (c) => {
+    const { embeddedUrl } = await getApp(
+      apiKey,
+      fetchFn,
+      fiberplaneServicesUrl,
+    );
+
+    const sessionKey = c.req.query("session");
+    if (!sessionKey) {
+      throw new HTTPException(403, { message: "Session key missing" });
+    }
+
+    setCookie(c, "fpSession", sessionKey);
+    return c.redirect(embeddedUrl);
+  });
+
+  app.post("/logout", async (c) => {
+    const sessionKey = getCookie(c, "fpSession");
+
+    if (sessionKey) {
+      // Remove session cookie from embedded app domain
+      deleteCookie(c, "fpSession");
+
+      // Invalidate session with auth server
+      await fetchFn(`${fiberplaneServicesUrl}/api/auth/invalidate-session`, {
+        method: "POST",
+        headers: {
+          Authorization: `Bearer ${apiKey}`,
+          "Content-Type": "application/json",
+          Cookie: `session=${sessionKey}`,
+        },
+      });
+    }
+
+    return c.json({ success: true });
+  });
```

Comment:
not yet implemented but this was my first pass at the logout api route inside the embedded